File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:



 bool isSubsequence(string s, string t) {

if(s=="")return true;
    
   int i=0;
   int j=0;

   while(i<t.length()){
    if(t[i]==s[j]){
        i++;
        j++;
    }
    else{
        i++;
    }
   }

   if(j<s.length()) return false;

   return true;













    }
};

File: 0392-is-subsequence/0392-is-subsequence.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function(s, t) {


    let sPointer = 0;

    let tPointer = 0;


    while(sPointer<s.length && tPointer<t.length){
        if(s[sPointer] === t[tPointer]){
            sPointer++;
            tPointer++;
        }
        else {
            tPointer++;
        }
    }


    if(sPointer===s.length) return true;




    return false;
    
};